### ISSUE TITLE
Cannot use project() objects in feature configurations

### ISSUE THREAD

**Issue by jvz** [lburgazzoli/gradle-karaf-plugin#14]
Here's what I'm trying to do. I want to make a configuration that corresponds to all my gradle projects that are being combined into a kar file. I'm not very familiar with the "Gradle way" of doing things, so what I tried was doing what I've done with the compile configuration:

```gradle
configurations {
  foo
}

dependencies {
  foo project(':module1'), project(':module2'), ...
}

karaf {
  features {
    // ...
    feature {
      name = 'foo'
      description = 'Foo project'
      version = '1.0.0'
      configurations = [project.configurations.foo]
    }
  }
}
```

When I try to run `gradle clean build`, I get an error saying that no file could be found named `module1/build/libs/module1-1.0.0.jar` which is true. This is because it's trying to resolve this file before the jar task has been run as far as I can tell.

I'm not sure if I'm just trying to use this wrong, so this could be an invalid bug. What I'm currently doing is using maven coordinates instead and running "gradle clean install build" instead of just "gradle build" which to me looks like an ugly workaround that I want to eliminate if possible.

**Comment by lburgazzoli** [comment]
You are absolutely right.

At the moment I'm refactoring/improving what was before include in gradle-karaf-features-plugin and not everything has been included.

**Comment by lburgazzoli** [comment]
I've released v0.0.18 that shoudl address this, can you please confirm ?

**Comment by jvz** [comment]
Yup, that version works as expected. The sha1 hashes also match between the build/libs/foo.jar and the version in the kar file. Thanks so much!
